lib/std/zig/system/NativePaths.zig

```diff
@@ -67,6 +71,8 @@ pub fn detect(arena: Allocator, native_target: std.Target) !NativePaths {
                 try self.addRPath(lib_path);
             } else if (mem.startsWith(u8, word, "-l")) {
                 // Ignore this argument.
+            } else if (mem.endsWith(u8, word, ".tbd")) {
+                // Ignore this argument.
```

Comment:
This should not be ignored, and actually handled in the linker. `tbd` is a textualised format for a dylib on macOS and skipping those may result in unexpected link failures.